### PR TITLE
AEA-1494 Combine Genus and Classification registration

### PR DESCRIPTION
**Pull request by 5A11**
This PR makes it possible to combine genus and classification registration steps.

[WIP]: Currently the PR only shows the solution in the generic_seller skill. Once approved, it will extend it to all other skills.

## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/fetchai/connections/soef/connection.py [hunk 1]
```diff
@@ -879,7 +879,13 @@ async def _set_personality_piece_handler(
         if not (isinstance(piece, str) and isinstance(value, str)):  # pragma: nocover
             raise SOEFException.debug("Personality piece bad values provided.")
 
-        await self._set_personality_piece(piece, value)
+        pieces = piece.split(",")
+        values = value.split(",")
+        if len(pieces) != len(values):
+            raise SOEFException.debug("Mismatched number of personality pieces and values.")
+
+        for idx in range(len(pieces)):
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/soef/connection.py:887]: use enumerate instead

File: packages/fetchai/connections/soef/connection.py [hunk 2]
```diff
@@ -879,7 +879,13 @@ async def _set_personality_piece_handler(
...
+            await self._set_personality_piece(pieces[idx], values[idx])
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/soef/connection.py:888]: Have you checked whether it is possible to set a whole (comma separated) list at once?

File: packages/fetchai/skills/generic_seller/behaviours.py [hunk 3]
```diff
@@ -148,8 +148,7 @@ def register_service_personality_classification(self) -> None:
         strategy = cast(GenericStrategy, self.context.strategy)
         descriptions = [
             strategy.get_register_service_description(),
-            strategy.get_register_personality_description(),
-            strategy.get_register_classification_description(),
+            strategy.get_registration_description(),
         ]
         oef_search_dialogues = cast(
             OefSearchDialogues, self.context.oef_search_dialogues
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/generic_seller/behaviours.py:154]: This loop below is a problem btw. It does not play well with the logic you submitted earlier this week. If both fail, then the first failed message can be overwritten with the second...

File: packages/fetchai/skills/generic_seller/behaviours.py [hunk 4]
```diff
@@ -148,8 +148,7 @@ def register_service_personality_classification(self) -> None:
         strategy = cast(GenericStrategy, self.context.strategy)
         descriptions = [
             strategy.get_register_service_description(),
-            strategy.get_register_personality_description(),
-            strategy.get_register_classification_description(),
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/generic_seller/behaviours.py:152]: I wasn't aware we already set these. So in some sense your PR isn't needed..it just offers a different way of doing it, which is still valuable in general.

File: packages/fetchai/skills/generic_seller/strategy.py [hunk 5]
```diff
@@ -190,6 +200,17 @@ def get_register_classification_description(self) -> Description:
         )
         return description
 
+    def get_registration_description(self) -> Description:
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/generic_seller/strategy.py:203]: The other two methods are no longer used and can be removed

File: packages/fetchai/skills/generic_seller/behaviours.py (hunk 3 above)
Comment (5A11) [reply packages/fetchai/skills/generic_seller/behaviours.py:154]: ok, the second bit can be broken down and moved into the SUCCESS and ERROR handlers

Comment (5A11) [reply packages/fetchai/skills/generic_seller/behaviours.py:154]: Will have to do that in all skills using the same pattern

File: packages/fetchai/skills/generic_seller/behaviours.py (hunk 4 above)
Comment (5A11) [reply packages/fetchai/skills/generic_seller/behaviours.py:152]: We do in some skills, not all. 

So task 1 is to combine the two registration steps, and task 2 is to essentially copy paste this solution into all skills.

File: packages/fetchai/connections/soef/connection.py (hunk 2 above)
Comment (5A11) [reply packages/fetchai/connections/soef/connection.py:888]: I have to see the SOEF api for that. Is there such a thing accessible?

Comment (5A11) [reply packages/fetchai/connections/soef/connection.py:888]: I had a look at a rough documentation.md file in the soef repo and the command doesn't seem to be getting multiple values (I may be mistaken as there was only some rough explanation there and nothing formal).

**Pull request closed without merging**